chain-signatures/node/src/protocol/signature.rs

```diff
@@ -552,6 +553,22 @@ impl SignatureManager {
                 .retry_exponential(10, 5)
                 .transact()
                 .await?;
+
+            let status = response.status();
+
+            // Check we returned successfully
+            let FinalExecutionStatus::SuccessValue(payload) = status else {
+                let err = format!("Publish transaction failed, {:?}", response);
+                tracing::error!(err);
+                return Err(near_fetch::Error::RpcReturnedInvalidData(err));
```

Comment:
We are publishing all signatures in a loop. Do we need to return an Error in case one of the transactions failed?

Reply:
We probably don't, we should probably log the error but not throw anything?

Reply:
That is an option. That is what we are doing now.
Ideally, we will need to add the signature back to the queue to try again in the next round. But only if the error is "recoverable", this one is not.

Reply:
Adding a signature back to the queue can create many edge cases and a need to track one more parameter - signature queue size. I would keep it simple for now and create a separate issue for this if you think it's important.

Reply:
also, don't need all this custom logic since we're using `near_fetch` here. We can just do
```
if let Err(err) = response.json() {
   ...
}
```
which does both the status checking and serialization step if needed

Reply:
Done